block/manager.go

```diff
@@ -184,23 +161,71 @@ func (m *Manager) Start(ctx context.Context) error {
 		// P2P Sync. Subscribe to P2P received blocks events
 		go uevent.MustSubscribe(ctx, m.Pubsub, "applyGossipedBlocksLoop", p2p.EventQueryNewGossipedBlock, m.onReceivedBlock, m.logger)
 		go uevent.MustSubscribe(ctx, m.Pubsub, "applyBlockSyncBlocksLoop", p2p.EventQueryNewBlockSyncBlock, m.onReceivedBlock, m.logger)
+		return nil
 	}
 
-	return nil
-}
-
-func (m *Manager) IsSequencerVerify() (bool, error) {
-	slProposerKey := m.SLClient.GetProposer().PublicKey.Bytes()
-	localProposerKey, err := m.LocalKey.GetPublic().Raw()
+	/* ----------------------------- sequencer mode ----------------------------- */
+	// Sequencer must wait till DA is synced to start submitting blobs
+	<-m.DAClient.Synced()
+	err = m.syncFromSettlement()
 	if err != nil {
-		return false, fmt.Errorf("get local node public key: %w", err)
+		return fmt.Errorf("sync block manager from settlement: %w", err)
 	}
-	return bytes.Equal(slProposerKey, localProposerKey), nil
+	// check if sequencer in the middle of rotation
+	nextSeqAddr, err := m.MissingLastBatch()
+	if err != nil {
+		return fmt.Errorf("missing last batch: %w", err)
+	}
+	// if sequencer is in the middle of rotation, complete rotation instead of running the main loop
+	if nextSeqAddr != "" {
+		m.handleRotationReq(ctx, nextSeqAddr)
+		return nil
+	}
```

Comment:
I dont' really see why special case needed here, if it's already handled by monitor sequencer rotation
seems wet and not needed

Reply:
it's to prevent the worker loops from running in this case
agree, probably can be improved. will be refactored anyway with the graceful role switch